src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,238 @@
+                + "(car_id, driver_id) VALUE (?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement insertDriversStatement =
+                        connection.prepareStatement(insertDriversRequest)) {
+            insertDriversStatement.setLong(1, car.getId());
+            for (Driver driver : car.getDrivers()) {
+                insertDriversStatement.setLong(2, driver.getId());
+                insertDriversStatement.executeUpdate();
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t insert drivers to " + car, e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE cars SET is_deleted = TRUE WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            return statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't delete car with id " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String getAllCarsForDriverRequest = "SELECT id, model "
+                + "FROM cars "
+                + "JOIN cars_drivers "
+                + "ON cars.id = cars_drivers.car_id "
+                + "WHERE cars_drivers.driver_id = ?;";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getAllCarsStatement =
+                        connection.prepareStatement(getAllCarsForDriverRequest)) {
+            getAllCarsStatement.setLong(1, driverId);
+            ResultSet resultSet = getAllCarsStatement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(parseCarsFromResultSet(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't find cars in DB by driver id=" + driverId, e);
+        }
+        for (Car car : cars) {
+            if (car != null) {
+                car.setDrivers(getDriversForCar(car.getId()));
+            }
+        }
+        return cars;
+    }
+
+    private Car parseCarsFromResultSet(ResultSet resultSet) throws SQLException {
+        Car car = new Car();
+        car.setId(resultSet.getObject("cars.id", Long.class));
+        car.setModel(resultSet.getString("cars.model"));
+        return car;
+    }
+
+    private Car getCar(ResultSet resultSet) throws SQLException {
+        Long id = resultSet.getObject("id", Long.class);
+        String model = resultSet.getString("model");
+        Long manufacturerId = resultSet.getObject("manufacturerId", Long.class);
+        Car car = new Car(id, model, manufacturerId);
+        car.setId(id);
+        return car;
+    }
```

Comment:
do we need both of these methods?

Reply:
I seem so... "parse.." take cars from "cars_drivers"DB, but "get..." takes from "cars"DB...

Reply:
I think it's better to take all car fields from db lets use getCar() method everywhere

Reply:
done